metaseq/checkpoint_utils.py

```diff
@@ -71,32 +67,15 @@ def is_better(a, b):
     suffix = trainer.checkpoint_suffix
     checkpoint_conds = collections.OrderedDict()
     checkpoint_conds[f"checkpoint{epoch}{suffix}.pt"] = (
-        end_of_epoch and not cfg.no_epoch_checkpoints and epoch % cfg.save_interval == 0
```

Comment:
i am early in review but now this reads like we will ALWAYS checkpoint at end of epoch.

Which is not the worst (it's actually kinda convenient for me) but hmmm. Want discussion from others on the team.

Reply:
It might make more sense to also check for `save_interval > 0`, similar to how it is done for `save_interval_updates`. This way you always have the option to turn of checkpointing by setting both `--save-interval` and `--save-interval-updates` to 0.

Reply:
Hm.. can we remove any possibility of checkpointing outside of the frequency of checkpoint interval (purely as a function of # of iterations) just to simplify for the time-being?

Reply:
I simplified further so that there is no possibility of checkpointing outside of the frequency of checkpoint interval.